infrastructure/lib/constructs/api-sqs-function.ts

```diff
@@ -0,0 +1,65 @@
+import * as iam from "@aws-cdk/aws-iam";
+import * as cdk from "@aws-cdk/core";
+import { HttpMethod } from "../http";
+import { ApiDynamoDBFunction, ApiDynamoDBFunctionProps } from "./api-dynamodb-function";
+import * as sqs from "@aws-cdk/aws-sqs";
+import { SqsEventSource } from "@aws-cdk/aws-lambda-event-sources";
+
+export interface ApiSQSFunctionProps extends ApiDynamoDBFunctionProps {
+  /**
+   * The SQS queue where data is sent
+   */
+  readonly queue: sqs.IQueue;
+  /**
+   * Optional param to create an SQS event source to recieve queue data
+   */
+  readonly createEventSource?: any;
+}
+
+/**
+ * Creates the required resources for a PortfolioDraft function for the ATAT API.
+ *
+ * The route is added to the API Gateway resource and the Lambda function is packaged and
+ * created. Additionally, the necessary permissions are granted on the DynamoDB table
+ * corresponding to the HTTP method specified in the props.
+ */
+export class ApiSQSFunction extends ApiDynamoDBFunction {
+  /**
+   * The SQS queue where data is sent
+   */
+  public readonly queue: sqs.IQueue;
+  /**
+   * Optional param to create an SQS event source to recieve queue data
+   */
+  public readonly createEventSource?: boolean;
```

Comment:
We need it to do: 
```
 if (props.createEventSource) {
      this.fn.addEventSource(new SqsEventSource(this.queue));
    }
```

Reply:
Not quite! We need it in the props (which is why you're doing `props.createEventSource`) but not on the class/object itself (which would allow `this.createEventSource`)

Reply:
![weird](https://user-images.githubusercontent.com/84199040/135882614-b73bbbbb-53c0-4300-ab31-2ca47c6695b6.PNG)